presto-delta/src/main/java/com/facebook/presto/delta/DeltaPageSourceProvider.java

```diff
@@ -394,4 +412,29 @@ else if (getParquetType(type, fileSchema, column, tableName, path).isPresent())
         }
         return getParquetType(prestoType, messageType, column, tableName, path);
     }
+
+    private static ColumnIndexStore getColumnIndexStore(Predicate parquetPredicate, ParquetDataSource dataSource, BlockMetaData blockMetadata, Map<List<String>, RichColumnDescriptor> descriptorsByPath, boolean readColumnIndexFilter)
```

Comment:
We talked earlier with @beinan about whether we should return optional or null for this methood. For this use case even we use optional, we still need to check .isPresent(). It is pretty much similar with null checking and not much value to do so. Here are some good explanations about null checking and Optional checking https://medium.com/javarevisited/null-check-vs-optional-are-they-same-c361d15fade3.

Reply:
get it. I think we try to use Optional as much as possible, trying to get rid of nullPointer exceptions. What do you think @beinan ?

Reply:
I am fine to replace it with Optional. But when I try it and see we still need to end up with null checking. This is because  ColumnIndexStore.java that defined in Parquet repo is using null checking.  So the class ParquetColumnIndexStore extends ColumnIndexStore need to have the same signature. So if we replace it with Optional. It ends up that we convert from null checking to Optional.empty() check in HDFSParquetDataSource and have to convert to Optiona.empty() to null checking in ParquetColumnIndexStore.  Since we only use empty() API in Optional, as that article mentioned, null checking and empty() is not much different in this case. Thoughts?

Reply:
get it. yep, bad Presto and Parquet are having different coding styles. 
how about we try Optional as much as possible in Presto code? This reminds me old days when try to implement new Parquet reader for Presto :)